ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -960,24 +937,22 @@ static bool TrySendRequestAndWaitForResponse(
     int* kauthResult,
     int* kauthError)
 {
+    // To be useful, the message needs to either provide an FSID/inode pair or a path
+    assert(vnodePath != nullptr || (vnodeFsidInode.fsid.val[0] != 0 || vnodeFsidInode.fsid.val[1] != 0));
     bool result = false;
+    const char* relativePath = nullptr;
     
     OutstandingMessage message =
     {
         .receivedResult = false,
         .rootHandle = root,
     };
     
-    if (nullptr == vnodePath)
+    if (nullptr != vnodePath)
```

Comment:
Should we still break if vnodePath == null?

Reply:
No, this is now explicitly a supported situation. If no path to the vnode is available, that's fine, we'll just need to rely on the fsid/inode. See the `assert()` a few lines up.

Reply:
@pmj that makes sense thanks!